frontend/src/components/contacts.tsx

```diff
@@ -0,0 +1,261 @@
+  const [showError, setShowError] = useState<boolean>(false);
+  // TODO T:110 (FK) Support adding addresses
+  // const [address, setAddress] = useState<string>('')
+
+  const handleChange =
+    (setState: React.Dispatch<React.SetStateAction<string>>) =>
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setState(event.target.value);
+    };
+  const handleFirstNameChange = handleChange(setFirstName);
+  const handleLastNameChange = handleChange(setLastName);
+  const handleEmailChange = handleChange(setEmail);
+  const handlePhoneNumberChange = handleChange(setPhoneNumber);
+  const handleNotesChange = handleChange(setNotes);
+  // const handleAddressChange = handleChange(setAddress);
+
+  const addRelationship = () => {
+    if (relationshipType.trim() === "") {
+      setShowError(true);
+      return;
+    }
+    setRelationships((prevRelationships) => [
+      ...prevRelationships,
+      {
+        type: relationshipType,
+      },
+    ]);
+    setRelationshipType("");
+  };
+
+  const handleRelationshipChange = (
+    index: number,
+    field: keyof Relationship,
+    value: any
+  ) => {
+    setRelationships((prevRelationships) =>
+      prevRelationships.map((type, i) =>
+        i === index ? { ...type, [field]: value } : type
+      )
+    );
+  };
+
+  const handleRemoveRelationshipType = (index: number) => {
+    setRelationships((prevRelationships) => [
+      ...prevRelationships.slice(0, index),
+      ...prevRelationships.slice(index + 1),
+    ]);
+  };
+
+  const addImportantEvent = () => {
+    if (eventDate.trim() === "" || eventType.trim() === "") {
+      setShowError(true);
+      return;
+    }
+    setImportantDates((prevDates) => [
+      ...prevDates,
+      {
+        date: eventDate,
+        event: eventType,
+      },
+    ]);
+    setEventDate("");
+    setEventType("");
+  };
+
+  const handleImportantEventChange = (
+    index: number,
+    field: keyof ImportantDate,
+    value: any
+  ) => {
+    setImportantDates((prevDates) =>
+      prevDates.map((date, i) =>
+        i === index ? { ...date, [field]: value } : date
+      )
+    );
+  };
+
+  const handleRemoveImportantEvent = (index: number) => {
+    setImportantDates((prevDates) => [
+      ...prevDates.slice(0, index),
+      ...prevDates.slice(index + 1),
+    ]);
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    try {
+      await axios.post("http://localhost:8000/user/contacts", {
+        firstName,
+        lastName,
+        relationships,
+        importantDates,
+        email,
+        phoneNumber,
+        notes,
+      });
+    } catch (error) {
+      console.error(error);
+      setErrorMessage((error as any).response.date ?? "Unknown error occured.");
+    }
+  };
+
+  return (
+    <div className="contacts_form">
+      <h1>Add a contact</h1>
+      <form action="/user/contacts" method="post" onSubmit={handleSubmit}>
```

Comment:
The action and method in not necessary here, the handleSubmit function will take care of your post request!
```suggestion
      <form onSubmit={handleSubmit}>
```